tinygrad/runtime/graph/hcq.py

```diff
@@ -51,7 +51,13 @@ def __init__(self, jit_cache: List[ExecItem], input_rawbuffers: List[Buffer], va
     for dev, queue in self.comp_queues.items(): dev_access[queue].add(dev)
 
     for j,ji in enumerate(self.jit_cache):
-      enqueue_dev = ji.prg.device if (is_exec_prg:=isinstance(ji.prg, CompiledRunner)) else Device[ji.bufs[1].device] #type:ignore
+      if not (is_exec_prg:=isinstance(ji.prg, CompiledRunner)):
+        assert ji.bufs[1] is not None, "can't select device"
+        enqueue_dev = Device[ji.bufs[1].device]
+      else: enqueue_dev = ji.prg.device
+      assert isinstance(enqueue_dev, HCQCompiled), "not instance of HCQCompiled"
+      assert enqueue_dev.hw_copy_queue_t is not None, 'no hw copy queue'
+      enqueue_queue:Union[HWComputeQueue, HWCopyQueue, HWCommandQueue]
```

Comment:
Is HwCommandQueue needed? Pay attention to inheritance

Reply:
Later in the code, enqueue_queue gets reassigned to a value of type HWCommandQueue. But inheritance allows switching this Union for just a HWCommandQueue type since the others inherit from it. I will make this change